src/btree/bt_cursor.c

```diff
@@ -927,6 +937,58 @@ __btcur_search_neighboring(WT_CURSOR_BTREE *cbt, WT_CURFILE_STATE *state, int *e
         if (*exact <= 0)
             return (ret);
     }
+
+    return (ret);
+}
+
+/*
+ * __btcur_search_neighboring --
+ *     Search for a valid record around the cursor location.
+ *
+ */
+static int
+__btcur_search_neighboring(
+  WT_CURSOR_BTREE *cbt, WT_CURFILE_STATE *state, bool prepare_conflict, int *exact)
+{
+    WT_DECL_RET;
+
+    /*
+     * If we haven't seen a prepared conflict error, walk forwards and then walk backwards.
+     *
+     * If we have seen a prepared conflict error, we must be reading with a snapshot. Therefore, no
+     * need to worry about the keys concurrently inserted and prepared as they are not visible to
+     * us.
+     */
+    if (!prepare_conflict) {
+        /*
+         * We didn't find an exact match: try after the search key, then before. We have to loop
+         * here because at low isolation levels, new records could appear as we are stepping through
+         * the tree.
+         */
```

Comment:
This comment starting from "We have to loop here because at low isolation levels" needs to be moved inside the `__btcur_search_walk_next` and `__btcur_search_walk_prev` functions